src/foam/dao/MongoDAO.java

```diff
@@ -112,8 +115,8 @@ public Sink select_(X x, Sink sink, long skip, long limit, Comparator order, Pre
     return sink;
   }
 
-  private FObject createFObject(X x, BsonDocumentReader reader, Class cls) {
-    FObject obj = (FObject) x.create(cls);
+  private FObject createFObject(X x, BsonDocumentReader reader, Class cls, NanoLogger logger) {
+    FObject obj = ( cls == null ) ? null : (FObject) x.create(cls);
```

Comment:
This was to handle the case when the mongo field name read in isn't recognized. In that case the value still needs to be read (and ignored) in order to ensure the parser state is maintained. `cls` can be null in the case the unknown field is a nested document. Here, if `cls` is null, it continues parsing through the nested document and just ignores the values.

The `cls` we care about would've been fetched from the DAO's `of` property on line 103.